base/frequencies.py

```diff
@@ -568,6 +570,164 @@ def test_nested_estimator():
     return nested_freq
 
 
+class KdeFrequencies(object):
+    """Methods to estimate clade frequencies for phylogenetic trees by creating
+    normal distributions from timestamped tips in the tree and building a kernel
+    density estimate across discrete time points from these tip observations for
+    each clade in the tree.
+    """
+    def __init__(self):
+        pass
+
+    @classmethod
+    def get_counts_from_observation(cls, mu, pivots, sigma=3 / 12.0, points=1000):
+        """Build a normal distribution centered across the given floating point date,
+        mu, with a standard deviation based on the given sigma value and map the
+        corresponding distribution into discrete bins defined by the given
+        pivots.
+
+        Returns counts and bin edges from a numpy histogram call where counts
+        represent a discretized normal distribution thay will compose a kernel
+        density estimate.
+        """
+        x = np.linspace(mu - 3 * sigma, mu + 3 * sigma, points)
+        y = norm.pdf(x, loc=mu, scale=sigma)
+
+        dt = pivots[1] - pivots[0]
+        bins = [pivots[0] - 0.5 * dt] + list(pivots + 0.5 * dt)
+        counts, bin_edges = np.histogram(x, bins=bins, weights=y, density=True)
+        return counts, bin_edges
+
+    @classmethod
+    def get_frequencies_from_observations(cls, observations, pivots, max_date=None, **kwargs):
+        """Create a matrix of frequencies for one or more observations across the given
+        pivots.
+
+        Observations can be optionally filtered by a maximum date such that all
+        frequencies are estimated to be zero after that date.
+        """
+        freq_matrix = np.zeros((len(observations), len(pivots)))
```

Comment:
Semantic, but easier for me to think of this as multiple frequency trajectories rather than a formal matrix. Perhaps just comment about what this is returning.